src/components/Switch/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { BaseProps } from '../types';
+import { SwitchWrapper, StyledSwitch, SwitchText } from './styled';
+import { COLOR_BRAND, COLOR_GRAY_2 } from '../../styles/colors';
+
+interface Props extends BaseProps {
+    label?: string;
```

Comment:
the label must be a ReactNode in order to support things like internationalization for example when using react intl we pass component that makes the translation